filebeat/harvester/reader/line.go

```diff
@@ -13,7 +13,7 @@ import (
 // using the configured codec. The reader keeps track of bytes consumed
 // from raw input stream for every decoded line.
 type Line struct {
-	rawInput   io.Reader
+	rawInput   io.ReadCloser
```

Comment:
same here, we can make `Close` optional using mentioned pattern.

something like

```
func ensureCloser(r io.Reader) io.ReadCloser {
    if rc, ok := r.(io.ReadCloser) {
      return rc
    }
    return iotuil.NopCloser(r)
}
```